ui.tests/test-module/specs/hcaptcha/hcaptcha.runtime.spec.js

```diff
@@ -0,0 +1,142 @@
+            Object.entries(formContainer._fields).forEach(([id, field]) => {
+                expect(field.getId()).to.equal(id)
+                expect(formContainer._model.getElement(id), `model and view are in sync`).to.equal(field.getModel())
+            });
+        }
+    })
+
+    it(" model's changes are reflected in the html ", () => {
+        if (cy.af.isLatestAddon() && toggle_array.includes(FT_HCAPTCHA)) {
+            const [id, fieldView] = Object.entries(formContainer._fields)[0]
+            const model = formContainer._model.getElement(id)
+            cy.get('div.h-captcha').should('exist');
+
+            checkHTML(model.id, model.getState()).then(() => {
+                model.visible = false
+                return checkHTML(model.id, model.getState())
+            }).then(() => {
+                model.enable = false
+                return checkHTML(model.id, model.getState())
+            })
+        }
+    });
+
+    it(" html changes are reflected in model ", () => {
+        if (cy.af.isLatestAddon() && toggle_array.includes(FT_HCAPTCHA)) {
+            const [id, fieldView] = Object.entries(formContainer._fields)[0]
+            const model = formContainer._model.getElement(id)
+            cy.log(model.getState().value)
+            cy.get(`#${id}`).click().then(x => {
+                cy.log(model.getState().value)
+                expect(model.getState().value).to.not.be.null
+            })
+        }
+    });
+
+
+    it("decoration element should not have same class name", () => {
+        if (cy.af.isLatestAddon() && toggle_array.includes(FT_HCAPTCHA)) {
+            expect(formContainer, "formcontainer is initialized").to.not.be.null;
+            cy.wrap().then(() => {
+                const id = formContainer._model._children[0].id;
+                cy.get(`#${id}`).parent().should("not.have.class", bemBlock);
+            })
+        }
+    })
+
+    it("client side validation should fail if captcha is not filled", () => {
+        if (cy.af.isLatestAddon() && toggle_array.includes(FT_HCAPTCHA)) {
+            expect(formContainer, "formcontainer is initialized").to.not.be.null;
+            cy.get(`.cmp-adaptiveform-button__widget`).click().then(x => {
+                cy.get('.cmp-adaptiveform-hcaptcha__errormessage').should('exist').contains('Please fill in this field.');
+            });
+        }
+    })
+
+    it("submission should pass for mandatory captcha", () => {
+        if (cy.af.isLatestAddon() && toggle_array.includes(FT_HCAPTCHA)) {
+            expect(formContainer, "formcontainer is initialized").to.not.be.null;
+            cy.get(`div.h-captcha iframe`).should('be.visible').then($iframe => {
+                cy.wrap($iframe).then($iframe => {
+                    cy.window().should('have.property', 'hcaptcha').and('not.be.undefined')
+                        .then((hcaptcha) => {
+                            hcaptcha.execute();
+                            cy.wait(2000);
```

Comment:
Lets not have a constant, instead use promise and wait for it to get resolved by giving a timeout